R/board_url.R

```diff
@@ -141,12 +205,100 @@ pin_store.pins_board_url <- function(board, name, paths, metadata,
   abort_board_read_only("board_url")
 }
 
+#' @export
+pin_version_delete.pins_board_url <- function(board, name, version, ...) {
+  abort_board_read_only("board_url")
+}
+
 #' @export
 write_board_manifest_yaml.pins_board_url <- function(board, manifest, ...) {
   abort_board_read_only("board_url")
 }
 
 # Helpers ------------------------------------------------------------------
+
+board_url_error_message <- c(
+  "{.var urls} must resolve to either:",
+  "*" = "an unnamed character scalar, i.e. a single URL",
+  "*" = "a named character vector",
+  "*" = "a named list, where all elements are character scalars or vectors"
+)
+
+get_url_format <- function(urls) {
+  if (is_scalar_character(urls) && !is_named(urls)) {
+    return ("pins_yaml")
+  }
+  if (is_list(urls) && is_named(urls) && all(map_lgl(urls, is_character))) {
+    return ("manifest_content")
+  }
+  if (is.character(urls) && is_named(urls)) {
+    return ("vector_of_urls")
+  }
+  cli::cli_abort(
+    board_url_error_message,
+    class = "pins_error_board_url_argument",
+    urls = urls
+  )
+}
+
+get_manifest <- function(url, call = rlang::caller_env()) {
+  # if ends with "/", look for manifest
+  if (grepl("/$", url)) {
+    url <- paste0(url, manifest_pin_yaml_filename)
+  }
+
+  # if request fails or returns with error code
+  tryCatch(
+    {
+      resp <- httr::GET(url)
+      httr::stop_for_status(resp)
+    },
+    error = function(e) {
+      cli::cli_abort(
+        message = c(
+          "Failed to access manifest file at {.url {url}}:",
+          " " = "{e$message}"
```

Comment:
Can remove this line since `cli_abort()` takes care of it because you've provided `parent`